### DISCUSSION TITLE
Need some help with my thought clarity on setting up SSR React with Parcel.

### DISCUSSION THREAD

**Discussion by abhinav-anshul** [parcel-bundler/parcel#9122] [Q&A]
Hi,

I am trying to set up a simple Server Side Rendered ReactJS App using react + express + parcel.

This is my `server.js` file

```
import express from "express"
import { renderToString } from "react-dom/server"
import React, { createElement } from "react"

const app = express()

app.get("/", (req, res, next) => {
  res.send(renderToString(<App />))
})

app.listen(3000, () => {
  console.log("RUNNING..")
})

```

Now the problem is -
```
SyntaxError: Unexpected token '<'
    at Loader.moduleStrategy (node:internal/modules/esm/translators:147:18)
    at async link (node:internal/modules/esm/module_job:64:21)
```

It does not read `<App />` component and i believe it wants me to pass vanilla javascript. So when i pass something like this -

```
  res.send(renderToString(createElement("h2", null, "Element rendered from Server")))
```
It runs perfectly and the HTML is being rendered on the screen. Now I would like to know -

1. I understand `renderToString` accepts a react tree as mentioned in the react doc. So why it throws an error when I pass `<App />` ? If I am simply passing js, then what the use of the `renderToString` API?
2. The second problem is - there are numerous articles that suggests to use transpilation of react components before passing to the `renderToString` using webpack as -

```
require("ignore-styles");

require("@babel/register")({
  ignore: [/(node_modules)/],
  presets: ["@babel/preset-env", "@babel/preset-react"],
});

require("./server");

```

Now, why do i need to transpile? Even so how can i achieve this using parcel only? I really do not want to use babel here.

Thanks for reading. If anyone can clear up my thought process on what I'm doing wrong?

**Comment by avoguga** [comment]
Hi there! Bro u r using the renderToString function from React that is designed to render React components into a string on the server. it expects a valid react element or component as its argument. In your case, <App /> is JSX syntax, which needs to be transpiled before it can be used in Node.js. The error you're seeing is because the server-side code does not recognize the JSX syntax.

If u want to try a SSR project with parcel and react see this: https://github.com/brillout/parcel-ssr

**Reply by abhinav-anshul** [reply]
Hi, thanks for the explanation. but isnt an `<App />` already a React tree? I believe a JSX component is a react tree already?

**Comment by abhinav-anshul** [comment]
Hi still need clarification on this
